src/opera/util/render_jinja2.py

```diff
@@ -109,7 +114,21 @@ def __getattr__(self, name):
     return LoggingUndefined
 
 
-def render_jinja2(template_filename: str, input_data: dict, logger: PgeLogger = None):
+def _try_parse_xml_string(s: str):
+    _ = etree.fromstring(s.encode('utf-8'))
+
+
+JSON_VALIDATOR = json.loads
+YAML_VALIDATOR = partial(yaml.load, Loader=yaml.SafeLoader)
+XML_VALIDATOR = _try_parse_xml_string
+
+
+def render_jinja2(
+        template_filename: str,
+        input_data: dict,
+        logger: PgeLogger = None,
+        validator=None
```

Comment:
I think I would rather see this parameter default to `XML_VALIDATOR` and have a caller decide to pass `None` if they want to disable validation. Then we won't need the changes to each individual PGE module in this branch.